digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/Point.scala

```diff
@@ -85,6 +85,10 @@ case class Vector3d(x: Double, y: Double, z: Double) {
   def to2D(): Vector3d = {
     Vector3d(x, y, 0.0)
   }
+
+  def angle(that: Vector3d): Double = {
```

Comment:
This function should be renamed to something else, since this is not calculating the angle between two 3d vectors as one would expect. Later someone might use this angle function and expect that also the z-component affects the calculation and that the possible return values would be between [0, π]. Now this implementation returns also negative angles and doesn't take z-component in account.

On the other hand if we fix this "angle" function implementation, the "pickRightMost"-function would probably be broken if it relies on this that the function returns also negative values.

Reply:
Created new method with the functionality that the pickRightMost function needs and changed this angle-method to do what I would expect it to do.